pkgs/by-name/ut/ut1999/package.nix

```diff
@@ -94,6 +99,11 @@ in stdenv.mkDerivation rec {
       runHook postInstall
     '';
 
+  # .so files in the SystemARM64 directory are not loaded properly on aarch64-linux
+  appendRunpaths = lib.optional (stdenv.hostPlatform.system == "aarch64-linux") [
+    "${placeholder "out"}/SystemARM64"
```

Comment:
Also perhaps you may want to generalise this a bit, by defining `systemDir` in a broader scope, and use `${systemDir}` here.